eZ/Publish/Core/Persistence/Legacy/User/Role/Gateway.php

```diff
@@ -1,170 +1,151 @@
 <?php
 
 /**
- * File containing the ContentTypeGateway class.
- *
  * @copyright Copyright (C) eZ Systems AS. All rights reserved.
  * @license For full copyright and license information view LICENSE file distributed with this source code.
  */
+declare(strict_types=1);
+
 namespace eZ\Publish\Core\Persistence\Legacy\User\Role;
 
 use eZ\Publish\SPI\Persistence\User\RoleUpdateStruct;
 use eZ\Publish\SPI\Persistence\User\Policy;
 use eZ\Publish\SPI\Persistence\User\Role;
 
 /**
- * Base class for content type gateways.
+ * User Role Gateway.
+ *
+ * @internal For internal use by Persistence Handlers.
  */
 abstract class Gateway
 {
+    public const ROLE_TABLE = 'ezrole';
+    public const POLICY_TABLE = 'ezpolicy';
+    public const POLICY_LIMITATION_TABLE = 'ezpolicy_limitation';
+    public const POLICY_LIMITATION_VALUE_TABLE = 'ezpolicy_limitation_value';
+    public const USER_ROLE_TABLE = 'ezuser_role';
+
+    public const ROLE_SEQ = 'ezrole_id_seq';
+    public const POLICY_SEQ = 'ezpolicy_id_seq';
+    public const POLICY_LIMITATION_SEQ = 'ezpolicy_limitation_id_seq';
+
     /**
-     * Create new role.
-     *
-     * @param Role $role
-     *
-     * @return Role
+     * Create a new role.
      */
-    abstract public function createRole(Role $role);
+    abstract public function createRole(Role $role): Role;
 
     /**
-     * Loads a specified role by $roleId.
+     * Load a specified role by $roleId.
      *
-     * @param mixed $roleId
      * @param int $status One of Role::STATUS_DEFINED|Role::STATUS_DRAFT
      *
      * @return array
      */
-    abstract public function loadRole($roleId, $status = Role::STATUS_DEFINED);
+    abstract public function loadRole(int $roleId, int $status = Role::STATUS_DEFINED): array;
 
     /**
-     * Loads a specified role by $identifier.
+     * Load a specified role by $identifier.
```

Comment:
Just curious, do we keep such redundant docblock description? Method name is: `loadRoleByIdentifier`.

Reply:
> Just curious, do we keep redundant docblock descriptions? Method name is: `loadRoleByIdentifier`.

It's rather per case basis. I still believe, looking at my PHPStorm, that human readable doc gives better DX then trying to parse method name in my head to understand what it does. Moreover some methods of that Gateway definitely require doc block, so not having them on some looks like a mistake/inconsistency.
But can remove that one if you really prefer it.